content/article/hotel-hotspot-hijinks.md

```diff
@@ -0,0 +1,300 @@
+```
+
+This gives us enough information to be able to submit the form using the
+relevant login data.
+
+Aside: interestingly enough, the fields are in English even though the site
+is a German one.  I guess standardising the fields on English can be useful
+when programming.
+
+To submit the form, we use `WWW::Mechanize`'s
+[`submit_form()`](https://metacpan.org/pod/WWW::Mechanize#$mech-%3Esubmit_form(-...-))
+method (the call to which I've formatted nicely here to make things easier
+to read):
+
+```perl
+$mech->submit_form(
+    form_name => 'login',
+    fields => {
+        username => 'username-for-room',
+        password => 'password-for-room',
+    }
+);
+```
+
+We can check if the form submission was successful by asking the
+[`HTTP::Response`](https://metacpan.org/pod/HTTP::Response) if things went
+well:
+
+```
+  DB<7> x $mech->res->is_success;
+0  1
+```
+
+Looking good so far.  Let's see if `ping` works as expected
+
+```
+$ ping www.heise.de
+PING www.heise.de (193.99.144.85) 56(84) bytes of data.
+64 bytes from www.heise.de (193.99.144.85): icmp_seq=1 ttl=247 time=20.6 ms
+64 bytes from www.heise.de (193.99.144.85): icmp_seq=2 ttl=247 time=13.9 ms
+^C
+--- www.heise.de ping statistics ---
+2 packets transmitted, 2 received, 0% packet loss, time 1000ms
+rtt min/avg/max/mdev = 13.941/17.281/20.621/3.340 ms
+```
+
+Yes!  In other words: we're in!  [That was
+easy](https://www.youtube.com/watch?v=kCE0v-SK5_c). &#x263A;
+
+Putting everything together (and cleaning up the code a bit), I ended up
+with this:
+
+```perl
+use strict;
+use warnings;
+
+use WWW::Mechanize;
+
+my $mech = WWW::Mechanize->new;
+$mech->get('http://google.com');
+
+# check that we have the login page and not Google or something else
+# i.e. we're not logged in
+if ($mech->title() =~ 'login') {
+    $mech->submit_form(
+        form_name => 'login',
+        fields => {
+            username => '<username-for-room>',
+            password => '<password-for-room>'
+        }
+    );
+    if ($mech->res->is_success) {
+        print "Login successful\n";
+    }
+    else {
+        print "Login failed\n";
+    }
+}
+else {
+    print "Already logged in\n";
+}
+```
+
+Now I can just run the script every time I wake my laptop back up and I'm
+back online.  Yay!
+
+From a security perspective it's a bit weird that the username and password
+are obviously tied to the room number.  In other words, I could probably
+use the neighbouring room's account just as easily (I guess: I couldn't
+be bothered checking in the end).
+
+## The conclusion
+
+Well, this script certainly saved me some time and hassle when waking up my
+laptop from the suspend state.  Also, it was fun working out what pieces of
+the puzzle were necessary in order to build a solution.  Perl saves the
+day[^perl-saves-the-day] again!
+
+[^perl-saves-the-day]: Well, it saved the week really.
+
+Originally posted on [https://peateasea.de](https://peateasea.de/hotel-hotspot-hijinks/).
```

Comment:
I'm glad you added this link. Hopefully it can drive some traffic over to your site.